### ISSUE TITLE
Downstream packaging of gedit is broken / still needs tepl 6.4.0_1 and not 6.5.1.

### ISSUE THREAD

**Issue by Frank-Zeyda** [Homebrew/homebrew-core#130859]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

Here the output of brew config:

```
HOMEBREW_VERSION: 4.0.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: c0f8068573bbf886fdc78d3f9daca54d53e7e49e
Last commit: 4 days ago
Core tap JSON: 12 May 17:43 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.Sw0GbvkyqF/org.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: N/A
Xcode: 14.3
Rosetta 2: false
```

And here the output of brew doctor:

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo frankzeyda/local-tap) checkout
```

Please not that the `local-tap` is currently empty - I set this up for some development and testing purposes only, and it does not interfere at all with the issue reported. Hence, please ignore this warning.

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Updating gedit for the GNOME text editor via `brew update/upgrade`, around 23 May 2023.

### What happened (include all command output)?

Since a homebrew core formulae update around 23 May 2023, `gedit` stopped working altogether for me (running the most recent Ventura 13.3.1 (1)). I.e., after starting gedit via the Terminal, the application produces some debug output and then simply hangs - no application window opens. Before that time, it already had become very unstable, i.e., around the time of March 2023, crashing frequently during startup and sometimes furthermore during use and termination (this appears to be an upstream issue though, please read on).

I reported and helped to debug those problems with the GNOME development team, please see the following Issue thread there:

https://gitlab.gnome.org/GNOME/gedit/-/issues/559

To summarize that discussion, the root cause for gedit **not starting** is that currently, gedit is packaged on Homebrew with version 6.5.1 of tepl, which is apparently still experimental and under development, and thus ought not be deployed. In order to package the most recent stable version of gedit (44.2 as per writing of this post), please use:

```
gedit 44.2
Tepl 6.4.0_1 (not 6.5.1 which is a development snapshot)
GtkSourceView 4.8.4 (not yet libgedit-gtksourceview, which is still in development too)
```

You'll find the formula for tepl 6.4.0_1 (bottle) here:

https://github.com/Homebrew/homebrew-core/commits/c013861dab4ec98e79bfe6bdff62aee09fc1537d?before=c013861dab4ec98e79bfe6bdff62aee09fc1537d+35&branch=c013861dab4ec98e79bfe6bdff62aee09fc1537d&path%5B%5D=Formula&path%5B%5D=tepl.rb&qualified_name=c013861dab4ec98e79bfe6bdff62aee09fc1537d

Now, to fix the problem on the brew side, tepl needs to be downgraded to 6.4.0_1 and the gedit.rb may need some adjustment too. If something else depends on tepl 6.5.1 (I assume this is not the case but may need to be checked) perhaps package tepl 6.4.0_1 and tepl 6.5.1 separately as different versions on Homebrew.

The second noted issues related to frequent crashes of `gedit` turns out to be independent of the brew packaging problem, so rather an upstream issue. I am discussing / proposing a fix with GNOME developers (see thread) and hope they can review this and roll out a new version of gedit 44 soon that amends that problem too. I.e., so that  gedit will become usable once again on macOS.

Lastly, it would be great if someone else using Ventura 13.3.1 (1) may still be able to confirm the above observations.

### What did you expect to happen?

The brew formulae update/upgrade around 23 May 2023 not to break the gedit application ... ;-)

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install gedit
```

or, if gedit is already installed, simply

```shell
brew update
brew upgrade
```

Then try to run `gedit` from the Terminal which should result in the application producing some debug output and then simply hanging (no seg. fault or the likes on my system, which is why it may have slipped through the CI/CD/CV pipeline).

**Comment by ZhongRuoyu** [comment]
We have downgraded `tepl` to 6.4.0. Please try:

```
brew update
brew upgrade
```

... and see if the issue related to `tepl` version is resolved.

The other `gedit` issue you mentioned is (as you mentioned) an upstream issue so we'll need the gedit team to fix it; there is not much we can do.

**Comment by Frank-Zeyda** [comment]
Hi Ruoyu,

Thanks very much for this. I tested the downgrade on my local MacBook and yes, tepl version 6.4.0_2 seems to have been pulled now and `gedit` starts up again (modulo the upstream issue). So, this is good progress!

As you said, we need to wait for the `gedit` team to fix the upstream bug. I already tried to help with this, see recent posts on:

https://gitlab.gnome.org/GNOME/gedit/-/issues/559

The baton is back to the GNOME team now, I can keep you posted here once there may be a new version roll-out of gedit 44.
